readers/flat_mutation_reader_v2.hh

```diff
@@ -174,6 +174,14 @@ public:
         void set_close_required() { _close_required = true; }
         static constexpr size_t default_max_buffer_size_in_bytes() { return 8 * 1024; }
 
+        virtual uint64_t get_validation_errors() {
```

Comment:
Hmm, the generic reader should not be changed for that.
Let's find another way to propagate these stats.

Reply:
would need more time for this one. All of my current ideas have significant drawbacks:

1. As like in current solution, it would require changes in generic reader. 
Class reader in which we are able to count the number of validation errors is an implementation of an abstract layer flat_mutation_reader_v2, thus direct access to its fields is impossible without changes in abstract layer (like in current solution).

2. We could pass a shared_ptr to scrub_compaction members and let reader update them directly (which I don't think is elegant), but make_sstable_reader() is const. Thus const_cast would need to be used or make_sstable_reader() would need to loose its const qualifier. Neither seems appropriate.

3. We could pass a reference (or pointer set to nullptr by default) to the counter through reader's methods, though again it would require changes in generic reader.

4. Playing with dynamic_cast and checking whether we deal with reader object would need to access private nested class reader, require changes in generic reader, and probably cause dependency problems.

I would need to dive deeper in the code, since currently I do not see any good solution. I would appreciate any tips.

Reply:
I did so, but then when I try to reach the counters from `compaction_scrub` I have to go through the abstract layer. That makes me write getters there and therefore also in `flat_mutation_reader_v2::impl`.

Reply:
See if you can store the counters in `compaction_data`.
@raphaelsc what do you think?

Reply:
I am not sure what it would change. `compaction_data` cannot be reached from the `flat_mutation_reader_v2` or `flat_mutation_reader_v2::impl`. Do you mean I should keep the counters there instead of `compaction_result`?

Also I have recently pushed the version that does not affect generic reader. Could you review it, please?

Reply:
Adding fields to the abstract interface to accommodate a single user is a no-go in the vast majority of cases.
I think (2) is the way to go. You can make your counter struct `mutable` and hence be able to pass non-const reference to the reader constructor (use const-cast if needed).